### ISSUE TITLE
Planned things

### ISSUE THREAD

**Issue by SuperSaiyajinStackZ** [Universal-Team/3DZwei#1]
Here's a list of planned things:
- [x] Support up to 20 Cardpairs. More probably won't be a thing because of screen space. Edit: Now supports up to 345 Pairs!
- [ ] Possibly some animations.
- [ ] NDS Version called `DSZwei`.
- [ ] Add MP3 support for Music and Sound Effects.
- [ ] Maybe T3X SpriteSheet generation within the app to allow custom cardset creation within the app. (This has the lowest priority though and is planned as the really last thing.)

You have Feature Requests? then feel free to comment under this issue. Stuff such as Online / Local play isn't planned yet, but might be for the future.

**Comment by SuperSaiyajinStackZ** [comment]
20 CardPair mode has been added! Next: Probably animations.

**Comment by SuperSaiyajinStackZ** [comment]
Increased max limit to 345 Pairs! Now the amount of pairs can be entered up from 1, to the max amount of pairs inside the cards spritesheet. 👀

**Comment by SuperSaiyajinStackZ** [comment]
Added Card Randomizer in the Settings Screen. Also updated planned things such as MP3 support for SFX & Music.. also T3X SpriteSheet generation within the app itself to allow Cardset generation without the need of a PC.

**Comment by SuperSaiyajinStackZ** [comment]
I'm gonna update this issue here, because of the coming Pull Request https://github.com/Universal-Team/3DZwei/pull/4.

- Local and Online Player for now isn't planned and will likely be cancelled.
- I don't really plan to support music in app honestly.
- Animations are already kinda implemented.
- Work on a NDS version might be worked on in the near future, because i don't think a memory cardgame as a homebrew app exist on the NDS yet so far.
- A Spritesheet generator in app is a pain and i decided to not add that.

Now to come to the things which the Pull Request does change / add:
- It now uses my new StackMem-Core which i wrote.
- Characters can now be customized with a t3x Spritesheet. (My 3DZwei-Sets UniStore will reflect the latest changes for sets after v0.3.0 is out, so Sets can be properly downloaded).
- Cardsets are now stored at `sdmc:/3ds/3DZwei/sets/cards/`.
- Character Sets are stored at `sdmc:/3ds/3DZwei/sets/characters/`.
- The UI changed a lot and it looks pretty good in my opinion. You can see the screenshots in the Rewrite branch.

This issue can likely be closed after the NDS version is alright.

**Comment by SuperSaiyajinStackZ** [comment]
The latest 2 commits in the Rewrite branch overhaul 3DZwei:

- Added fancy animations, such as card flips, cubic bezier etc.
- Get rid of screen classes and only use Overlays.
- Using system keyboard now because it's better.
- Changed set path to cards: `sdmc:/3ds/ut-games/sets/3DZwei/` and characters: `sdmc:/3ds/ut-games/sets/characters/`.

And more.. just see the changes i guess.

**Comment by SuperSaiyajinStackZ** [comment]
The Rewrite has been merged into master now. The changes are now available in the latest nightly version.

**Comment by Epicpkmn11** [comment]
NightScript suggestions:
- [ ] First off, I think it should be a vertical slide rather than a horizontal slide
- [ ] Second off, instead of doing pages for the cards in the BG, why not do a continuous scrolling BG
Something like Sonic Manias "thanks for playing demo" screen

Universal-Tambourine suggestions:
- [ ] What if instead of having it do all those pages, if it just animated the entire page scroll at once
Like the whole 20 or so pages all in one go
  - Evie: So like normal speed for like up to 5 or so, then fast up to 10 or 15 or so, then normal speed but direct to the destination above that or so
- [ ] And then maybe similar for clicking the scroll button really fast

Evie suggestions:
- [x] Might want to speed up consecutive page moves by the AI, it's a bit slow lol
- [x] Also imo there should be some way to quit the game by default, it's annoying if you get the settings wrong and didn't set a hotkey yet ~~especially in Citra where it doesn't like to save~~
  Maybe just like make START the default hotkey or something
  ```
    Are you sure you want to quit?
             [No]    [Yes]
  ```
- [x] Make player 2 default to the second avatar
- [ ] When there are 3 or 4 pairs the first card flies in twice ([video](https://cdn.discordapp.com/attachments/734878807223304242/845865279066341386/o.mp4))
- [ ] Hmm, another thing that feels a bit unpolished to me is the intro splash. It's a bit odd how the things aren't centered when it pauses ([video](https://cdn.discordapp.com/attachments/734878807223304242/846112605953458256/2021-05-23_14.46.17.mov))
  - I think it might be better to start with StackMemCore centered (and Universal-Core fully off screen)
  - Then scroll so that Universal-Core is centered and StackMemCore is fully off screen
  - Then either finish the same or with just your picture centered but wait on that for a second or so
  - The cursor should probably also either move a bit faster or move diagonally imo
  - There's also no acceleration at all in those animations
- [x] Make Circle Pad analog
  - Note: Adjustments to deadzone / speed might be needed with more testing
- [x] Make "Guess" count be total guesses and add "Miss" count for just the incorrect guesses
  - Strings are already added, just needs to be added to the code
- [ ] Make buttons able to interact with the quit game prompt, probably just allow the cursor to be used though it could be <kbd>A</kbd> to quit and <kbd>B</kbd> to cancel
